.ci/azure-pipelines/release.yaml

```diff
@@ -0,0 +1,154 @@
+# Release PCL steps:
+# 1. Check tests:
+#   * debian:latest
+#   * perception_pcl
+# 2. Package for
+#   * MacOS
+#   * tar source release
+#   * Windows Installers
+
+trigger: none
+
+resources:
+- repo: self
+
+variables:
+  dockerHub: "PersonalDockerHub"
+  dockerHubID: "kunaltyagi"
+
+stages:
+- stage: Debian
+  dependsOn: []
+  jobs:
+  - job: BuildDebian
+    displayName: "Build Debian Latest"
+    timeoutInMinutes: 360
+    pool:
+      vmImage: 'ubuntu-latest'
+    variables:
+      tag: "release"
+    steps:
+    - checkout: self
+      # find the commit hash on a quick non-forced update too
+      fetchDepth: 10
+    - task: Docker@2
+      displayName: "Build docker image"
+      inputs:
+        command: build
+        arguments: |
+          --no-cache
+          -t $(dockerHubID)/release:$(tag)
+        dockerfile: '$(Build.SourcesDirectory)/.dev/docker/release/Dockerfile'
+        tags: "$(tag)"
+- stage: ROS
+  dependsOn: []
+  jobs:
+  - job: PerceptionPCL
+    displayName: "perception_pcl compile"
+    timeoutInMinutes: 360
+    pool:
+      vmImage: 'ubuntu-latest'
+    strategy:
+      matrix:
+        # Only ROS Melodic works with current releases of PCL
+        # Kinetic has stopped supporting any version later than PCL 1.7.2
+        ROS Melodic:
+          flavor: "melodic"
+    variables:
+      tag: "ros"
+    steps:
+    - checkout: self
+      # find the commit hash on a quick non-forced update too
+      fetchDepth: 10
+    - task: Docker@2
+      displayName: "Build docker image"
+      inputs:
+        command: build
+        arguments: |
+          --no-cache
+          --build-arg flavor=$(flavor)
+          -t $(dockerHubID)/release:$(tag)
+        dockerfile: '$(Build.SourcesDirectory)/.dev/docker/perception_pcl_ros/Dockerfile'
+        buildContext: '$(Build.SourcesDirectory)/.dev/docker/perception_pcl_ros'
+        tags: "$(tag)"
+- stage: Prepare
+  dependsOn: ["Debian", "ROS"]
+  jobs:
+  - job: src_code
+    displayName: "Package Source Code"
+    timeoutInMinutes: 360
+    pool:
+      vmImage: 'ubuntu-latest'
+    variables:
+      SOURCE_TAR: '$(Build.ArtifactStagingDirectory)/source.tar.gz'
+    steps:
+    - checkout: self
+      # find the commit hash on a quick non-forced update too
+      fetchDepth: 10
+    - task: Bash@3
+      displayName: "Remove git files"
+      inputs:
+        targetType: 'inline'
+        script: 'rm -fr ./.git'
+        workingDirectory: '$(Build.SourcesDirectory)'
+        failOnStderr: true
```

Comment:
You can pass `--exclude-vcs` to tar and eliminate this step.

Reply:
AFAIK, option is not available: https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#checkout

Alternative is to `checkout: none` and use a raw command

Reply:
Ignore